runners/core-java/src/test/java/org/apache/beam/runners/core/ReduceFnRunnerTest.java

```diff
@@ -351,6 +389,38 @@ public BoundedWindow window() {
   }
 
   /**
+   * Tests that when a processing time timers comes in after a window is expired
```

Comment:
Is it worth having one that tests that when a processing time timer comes in with a timestamp > GC time, it doesn't *cause* GC (mostly asking given that was one of the bugs, that isn't explicitly tested here).